code/game/gamemodes/nuclear/nuclear_challenge.dm

```diff
@@ -23,24 +24,37 @@
 		return
 
 	if(world.time > CHALLENGE_TIME_LIMIT)
-		user << "It's too late to declare hostilities. Your benefactors are already busy with other schemes. You'll have to make  do with what you have on hand."
+		user << "It's too late to declare hostilities. Your benefactors are already busy with other schemes. You'll have to make do with what you have on hand."
 		return
 
 	if(used) //First used check
 		return
 
-	var/are_you_sure = alert(user, "Consult your team carefully before you declare war on [station_name()]]. Are you sure you want to alert the enemy crew?", "Declare war?", "Yes", "No")
+	var/are_you_sure = alert(user, "Consult your team carefully before you declare war on [station_name()]. Are you sure you want to alert the enemy crew? You have [round((CHALLENGE_TIME_LIMIT - world.time)/10)] seconds to decide", "Declare war?", "Yes", "No")
 	if(are_you_sure == "No")
 		user << "On second thought, the element of surprise isn't so bad after all."
 		return
 
 	if(used) //Second used check incase it's sustained in the dialog
 		user << "You already declared war on the station!"
 		return
+	
+	var/war_declaration = "[user.real_name] has declared his intent to utterly destroy [station_name()] with a nuclear device, and dares the crew to try and stop them."
+	var/custom_threat = alert(user, "Do you want to customize your declaration?", "Customize?", "Yes", "No")
+	if(custom_threat == "Yes" && (world.time < CHALLENGE_TIME_LIMIT-600))
+		war_declaration = copytext(sanitize(input(user, "Insert your custom declaration", "Declaration")as text | null), 1, 500)
```

Comment:
https://github.com/HippieStationCode/HippieStation13/blob/b149a510ae9aa5a2ea4339f59c14b69b42794bdb/code/__HELPERS/text.dm#L77
this proc fits better